### DISCUSSION TITLE
Some styles not being applied in Next.js TS project

### DISCUSSION THREAD

**Discussion by NinaSkyttis** [tailwindlabs/tailwindcss#14213] [Help] [answered]
Hi there, 

Excuse me if this is a ridiculous question, but it can't possibly be because I've spent several hours with a friend of mine trying to figure out why some Tailwind classes won't apply to my project. This is a coding challenge that was sent to me by a potential employer. The goal is to build three components mimicking their Figma files exactly, using next.js, TyoeScript and Tailwind. 

The project itself isn't that complicated, but once I'm trying to use Tailwind, only classes like background color and text color, don't size etc are being applied. Apart from that I'm basically not able to apply any other classes (like margin and padding) and it's driving me a bit nuts. I've looked into if:
- any other classes are overriding
- It would work on another computer
- Uncommenting their initial SCSS styling 
- Asked ChatGPT 
- Googled 

Maybe I'm missing something very obvious, but I would so greatly appreciate if anyone would be willing to help me! 

Thank you so much, 
Nina

**Comment by wongjn** [comment]
Consider providing a project that reproduces the unexpected behaviour.

**Reply by NinaSkyttis** [reply]
Hey! Thanks for your reply. I invited you to the repository. Inside of the file /components/blocks/text_image/text_image.tsx I've created a very simple button for reference. The border-2 is being applied, but not the padding. 

<img width="234" alt="example_tailwind" src="https://github.com/user-attachments/assets/9bfa41aa-73d6-4c8a-b569-b174156c23cc">

**Reply by wongjn** [reply, marked as answer]
You've defined some custom spacing in `theme.spacing`:

```js
theme: {
  spacing: {
    0: '0px',
    10: '1rem',
  },
```
which means that the [default spacing scale](https://tailwindcss.com/docs/customizing-spacing#default-spacing-scale) is no longer loaded into Tailwind. This is because by defining values in `theme` (as opposed to [`theme.extend`](https://tailwindcss.com/docs/theme#extending-the-default-theme)) [stops the default values from Tailwind being loaded](https://tailwindcss.com/docs/theme#overriding-the-default-theme). The padding utilities take their values from the `spacing` values, thus the only valid `px-*` classes would be `px-0` or `px-10`.

**Reply by NinaSkyttis** [reply]
I can't believe it, Thank you so so so much. I should have figured it out on my own but I can't tell you how much I appreciate this. THANK YOU!
